applications/GeoMechanicsApplication/tests/cpp_tests/test_axisymmetric_stress_state.cpp

```diff
@@ -0,0 +1,108 @@
+#include "testing/testing.h"
+#include <boost/numeric/ublas/assignment.hpp>
+
+using namespace Kratos;
+
+namespace Kratos::Testing
+{
+
+ModelPart& CreateModelPart(Model& rModel)
+{
+    ModelPart& result = rModel.CreateModelPart("Main");
+
+    result.CreateNewNode(1, 0.0, 0.0, 0.0);
+    result.CreateNewNode(2, 1.0, 0.0, 0.0);
+    result.CreateNewNode(3, 1.0, 1.0, 0.0);
+
+    std::vector<ModelPart::IndexType> node_ids{1, 2, 3};
+    result.CreateNewElement("UPwSmallStrainElement2D3N", 1, node_ids, result.CreateNewProperties(0));
+
+    return result;
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateBMatrixWithValidGeometryReturnsCorrectResults, KratosGeoMechanicsFastSuite)
+{
+    const auto p_stress_state_policy = std::make_unique<AxisymmetricStressState>();
+
+    Model      model;
+    ModelPart& model_part = CreateModelPart(model);
+
+    Vector Np(3);
+    Np <<= 1.0, 2.0, 3.0;
+
+    // clang-format off
+    Matrix GradNpT(3, 2);
+    GradNpT <<= 1.0, 2.0,
+                3.0, 4.0,
+                5.0, 6.0;
+    // clang-format on
+
+    const Matrix calculated_matrix =
+        p_stress_state_policy->CalculateBMatrix(GradNpT, Np, model_part.GetElement(1).GetGeometry());
+
+    // clang-format off
+    Matrix expected_matrix(4, 6);
+    expected_matrix <<= 1   ,0  ,3   ,0 ,5   ,0, // This row contains the first column of GradNpT on columns 1, 3 and 5
+                        0   ,2  ,0   ,4 ,0   ,6, // This row contains the second column of GradNpT on columns 2, 4 and 6
+                        0.2 ,0  ,0.4 ,0 ,0.6 ,0, // This row contains Np/radius on columns 1, 3 and 5, where radius = 5
+                        2   ,1  ,4   ,3 ,6   ,5; // This row contains the first and second columns of GradNpT, swapping x and y
+    // clang-format on
+
+    KRATOS_CHECK_MATRIX_NEAR(calculated_matrix, expected_matrix, 1e-12)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ReturnCorrectIntegrationCoefficient, KratosGeoMechanicsFastSuite)
+{
+    const auto p_stress_state_policy = std::make_unique<AxisymmetricStressState>();
+
+    Model      model;
+    ModelPart& model_part = CreateModelPart(model);
+
+    // The shape function values for this integration point are 0.2, 0.5 and 0.3 for nodes 1, 2 and 3 respectively
+    Geometry<Node>::IntegrationPointType integration_point(0.5, 0.3, 0.0, 0.5);
+
+    const double detJ                   = 2.0;
+    const double calculated_coefficient = p_stress_state_policy->CalculateIntegrationCoefficient(
+        integration_point, detJ, model_part.GetElement(1).GetGeometry());
+
+    // The expected number is calculated as follows:
+    // 2.0 * pi * 0.8 (radius) * 2.0 (detJ) * 0.5 (weight) = 5.02655
+    KRATOS_EXPECT_NEAR(calculated_coefficient, 5.02655, 1e-5);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TestCloneReturnsCorrectType, KratosGeoMechanicsFastSuite)
+{
+    const auto p_stress_state_policy        = std::make_unique<AxisymmetricStressState>();
+    const auto p_cloned_stress_state_policy = p_stress_state_policy->Clone();
```

Comment:
The idea of the `Clone` member function is to make a copy of the object that is behind a `StressStatePolicy` reference or pointer. We can't use the copy constructor in that case, since it would slice the referenced object. To prove that the `Clone` member function works as intended, I would introduce a `const StressStatePolicy&` and call `Clone()` on that reference. That is:
```suggestion
    const StressStatePolicy& r_policy_base  = *p_stress_state_policy;
    const auto p_cloned_stress_state_policy = r_policy_base.Clone();
```

Reply:
Agreed, I made it more clear now that the p_stress_state_policy is a `std::unique_ptr<StressStatePolicy>`  and did the same for the other tests, because that's how we are planning to use it.